repos/keg-cli/src/utils/services/publishService.js

```diff
@@ -0,0 +1,335 @@
+const { git } = require('KegGitCli')
+const { Logger } = require('KegLog')
+const { spawnCmd } = require('KegProc')
+const { ask } = require('@keg-hub/ask-it')
+const { copySync, emptyDirSync } = require('KegFileSys/fileSys')
+const { get } = require('@keg-hub/jsutils')
+const { getHubRepos } = require('../hub/getHubRepos')
+const { versionService } = require('./versionService')
+const { generalError } = require('../error/generalError')
+const { runRepoScript } = require('../hub/runRepoScript')
+const { getPublishContext } = require('../publish/getPublishContext')
+const { getPublishContextOrder } = require('../publish/getPublishContextOrder')
+const { getVersionUpdate, getValidSemver } = require('KegUtils/version')
+/**
+ * Attempts to rollback changes made to a repo using git
+ * @function
+ * @param {Object} repo - Repo to be rolled back
+ * @param {Object} publishArgs - Define the state or the repo being published
+ *
+ * @returns {Void}
+ */
+const rollbackChanges = async (repo, publishArgs) => {
+  const { currentBranch, newVersion, wasPublished, step } = publishArgs
+
+  logFormal(repo, `Publish service failed on step ${step[1]}!\nRolling back publish changes...`)
+
+  const doGitReset = await ask.confirm(`Confirm running a full git reset. ALL CHANGES WILL BE LOST`)
+  if(!doGitReset) return Logger.warn(`Canceling git reset. Rollback did not complete. Current git branch is not clean!`)
+
+  if(step[0] > 3 && wasPublished)
```

Comment:
I'm going to update the `step` property to be an object with `number` and `name` instead of it being an array. hopefully that will make it more clear.

it's suppose to say, if we're on step 4 or above, do such and such

Reply:
thanks, much clearer now to me